packages/tests/tests/filter/multiple_pubsub.node.spec.ts

```diff
@@ -178,3 +180,177 @@ describe("Waku Filter V2: Multiple PubsubTopics", function () {
     }
   });
 });
+
+describe("Waku Filter V2 (Autosharding): Multiple PubsubTopics", function () {
+  // Set the timeout for all tests in this suite. Can be overwritten at test level
+  this.timeout(30000);
+  let waku: LightNode;
+  let nwaku: NimGoNode;
+  let nwaku2: NimGoNode;
+  let subscription: IFilterSubscription;
+  let messageCollector: MessageCollector;
+
+  const customContentTopic1 = "/waku/2/content/test.js";
+  const customContentTopic2 = "/myapp/1/latest/proto";
+  const autoshardingPubsubTopic1 = contentTopicToPubsubTopic(
+    customContentTopic1,
+    3
+  );
+  const autoshardingPubsubTopic2 = contentTopicToPubsubTopic(
+    customContentTopic2,
+    3
+  );
+  const contentTopicInfo: ContentTopicInfo = {
+    clusterId: 3,
+    contentTopics: [customContentTopic1, customContentTopic2]
+  };
+  const customEncoder1 = createEncoder({
+    contentTopic: customContentTopic1,
+    pubsubTopicShardInfo: {
+      clusterId: 3
+    }
+  });
+  const customDecoder1 = createDecoder(customContentTopic1, { clusterId: 3 });
+  const customEncoder2 = createEncoder({
+    contentTopic: customContentTopic2,
+    pubsubTopicShardInfo: {
+      clusterId: 3
+    }
+  });
+  const customDecoder2 = createDecoder(customContentTopic2, { clusterId: 3 });
+
+  this.beforeEach(async function () {
+    this.timeout(15000);
+    [nwaku, waku] = await runNodes(
+      this,
+      [autoshardingPubsubTopic1, autoshardingPubsubTopic2],
+      contentTopicInfo
+    );
+    subscription = await waku.filter.createSubscription(
+      pubsubTopicToSingleShardInfo(autoshardingPubsubTopic1)
+    );
+    messageCollector = new MessageCollector();
+  });
+
+  this.afterEach(async function () {
+    this.timeout(15000);
+    await tearDownNodes([nwaku, nwaku2], waku);
+  });
+
+  it("Subscribe and receive messages on autosharded pubsubtopic", async function () {
+    await subscription.subscribe([customDecoder1], messageCollector.callback);
+    await waku.lightPush.send(customEncoder1, { payload: utf8ToBytes("M1") });
+    expect(
+      await messageCollector.waitForMessagesAutosharding(1, {
+        contentTopic: customContentTopic1
+      })
+    ).to.eq(true);
+    messageCollector.verifyReceivedMessage(0, {
+      expectedContentTopic: customContentTopic1,
+      expectedPubsubTopic: autoshardingPubsubTopic1,
+      expectedMessageText: "M1"
+    });
+  });
+
+  it("Subscribe and receive messages on 2 different pubsubtopics", async function () {
+    await subscription.subscribe([customDecoder1], messageCollector.callback);
+
+    // Subscribe from the same lightnode to the 2nd pubsubtopic
+    const subscription2 = await waku.filter.createSubscription(
+      pubsubTopicToSingleShardInfo(autoshardingPubsubTopic2)
+    );
```

Comment:
This API does not look great. possibly a future improvement is that the user just pass their content topic here.

Reply:
opened an issue for this https://github.com/waku-org/js-waku/issues/1764